R-package/NAMESPACE

```diff
@@ -32,6 +32,10 @@ export(slice)
 export(xgb.DMatrix)
 export(xgb.DMatrix.hasinfo)
 export(xgb.DMatrix.save)
+export(xgb.DataIter)
+export(xgb.ExternalDMatrix)
```

Comment:
Let's not expose this as an independent class for now. One day we might support external storage with QuantileDMatrix.

What do you think if we make the `data` parameter of `DMatrix` to be compatible with the iterator?

Reply:
That makes sense, I have been struggling with this since the first review and have been stalling the decision (my bad). Will look closer in the new class.